### PR TITLE
add fold when rewriting constant expressions

### PR DESCRIPTION
**Pull request by jn80842**

### CODE REVIEW

**Comment by shoaibkamil** [comment]
@jn80842 Can you rebase to the current halide:metaprogrammed_simplifier_rules to get rid of the conflicts?

**Comment by jn80842** [comment]
Okay, merged.

File: src/Simplify_Add.cpp [hunk 1]
```diff
@@ -79,9 +79,15 @@ Expr Simplify::visit(const Add *op, ConstBounds *bounds) {
                rewrite(x*y + x, x * (y + 1)) ||
                rewrite(y*x + x, (y + 1) * x, !is_const(x)) ||
                rewrite((x + c0)/c1 + c2, (x + fold(c0 + c1*c2))/c1) ||
+<<<<<<< HEAD
```
Comment (shoaibkamil) [inline src/Simplify_Add.cpp]: I think you forgot to fix the merge conflict?

**Comment by jn80842** [comment]
Eek, fixed.

**Pull request merged**